src/Core/View.js

```diff
@@ -64,10 +63,21 @@ function _preprocessLayer(view, layer, parentLayer) {
         _syncGeometryLayerVisibility(layer, view);
         // Find projection layer, this is projection destination
         layer.projection = view.referenceCrs;
-    } else if (parentLayer.tileMatrixSets.includes(CRS.formatToTms(source.projection))) {
+    } else if (parentLayer && parentLayer.tileMatrixSets && parentLayer.tileMatrixSets.includes(CRS.formatToTms(source.projection))) {
         layer.projection = source.projection;
     } else {
-        layer.projection = parentLayer.extent.crs;
+        layer.projection = parentLayer && parentLayer.extent.crs;
+    }
+
+    if (layer.labelEnabled && source) {
+        const labelLayer = new LabelLayer(`${layer.id}-label`, view.referenceCrs);
+        labelLayer.source = source;
+        labelLayer.style = layer.style;
+
+        layer.whenReady = layer.whenReady.then(() => {
+            view.addLayer(labelLayer);
+            return layer;
+        });
```

Comment:
could you try `view.addLayer(labelLayer);` only?

Reply:
It works, thanks !